### ISSUE TITLE
Command Line Interface (CLI)

### ISSUE THREAD

**Issue by overheadhunter** [cryptomator/cli#1]
_From @SeameX on March 9, 2015 18:8_

Hello,

it would be awesome if there could be a command line version!
I want to crypt files on my rpi without GUI.

Best regards

SeameX

_Copied from original issue: cryptomator/cryptomator#43_

**Comment by overheadhunter** [comment]
In principal this will be possible ;-) UI is decoupled from crypto and webdav implementation.

But I guess you have to be patient, as I don't want to start a second frontend before the interfaces to the core functionality are final.

**Comment by overheadhunter** [comment]
_From @philosowaffle on September 25, 2015 19:1_

+1 for this feature

**Comment by overheadhunter** [comment]
_From @thiloms on February 10, 2016 10:10_

+1 also for this feature

**Comment by overheadhunter** [comment]
_From @opensource21 on May 15, 2016 20:9_

+1 it's important to use it in backup-scripts

**Comment by overheadhunter** [comment]
_From @Meresmata on June 1, 2016 10:4_

+1, yes that would be very usefull...

**Comment by overheadhunter** [comment]
_From @hypatia1 on June 3, 2016 11:51_

A CLI would make Cryptomator much more useful

**Comment by overheadhunter** [comment]
_From @abaumfalk on June 3, 2016 19:16_

+1 - would like to write an unlocking script for personal use, that receives the key from an usb drive; cli is urgently needed for this...

**Comment by overheadhunter** [comment]
_From @chrootLogin on July 11, 2016 8:49_

+1 for this feature... It would make cryptomator more awesome!

**Comment by overheadhunter** [comment]
_From @xelra on July 28, 2016 20:48_

I also want to voice my interest in this feature. A CLI interface is absolutely essential. I cannot use Cryptomator for any serious deployments without it, since all my servers are CLI only. I was just in the process of setting up an encryption-sync-backup scheme and then had to realize that there's no CLI. I'll have to put my plans on hold for now.

**Comment by overheadhunter** [comment]
_From @derekwolfson on August 5, 2016 17:52_

Also +1.  This would absolutely rock for my workflows. 

As an additional feature it'd be nice to allow a CL call w/o a password and have cryptomator prompt for a password.

I am writing code where we are analyzing encrypted data, would be so nice to just share the password with those who need it and have them install cryptomator and then they just need to enter the password at the prompt in the CL to access the data.

**Comment by overheadhunter** [comment]
_From @Xandarian on August 16, 2016 17:49_

+1 for this feature. I need it a lot!

**Comment by overheadhunter** [comment]
_From @crimsdings on September 28, 2016 10:40_

+1

**Comment by overheadhunter** [comment]
_From @dodekeract on September 28, 2016 23:26_

Can you guys stop writing +1? There's a feature that lets you add a `:+1:` to the issue's original comment, thanks.

**Comment by overheadhunter** [comment]
_From @jtomaszewski on November 9, 2016 22:33_

How about we'd use a WebDAV CLI to copy/save files in the volume? Is there any?

Then the only remaining command I'd need then is to mount the volume, either through CLI or through automount feature (https://github.com/cryptomator/cryptomator/issues/40).

After that, I could fully automate my deployment setup to work with cryptomator - which would be awesome ;]

**Comment by overheadhunter** [comment]
Good news, everyone! 🎉 

I have just successfully integrated two of our externalized libraries, [CryptoFS](https://github.com/cryptomator/cryptofs), which is a NIO filesystem implementation that does all the crypto stuff, and [webdav-nio-adapter](https://github.com/cryptomator/webdav-nio-adapter) which surprisingly provides WebDAV access to any NIO path.

Will publish these experiments within the next few days on GitHub. I decided an appropriate name for this new project might be cryptomator-cli 😉 

Stay tuned!
